.github/workflows/build-and-test.yaml

```diff
@@ -0,0 +1,38 @@
+name: Build and Test
+run-name: Build and Test 🚀
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-22.04
+    container: fedora:38
+    strategy:
+      fail-fast: false
+      matrix:
+        buildtype:
+          - Debug
+          - Release
+    steps:
+      - name: Install packages
+        run: |
+          sudo dnf -y install 'dnf-command(copr)'
+          sudo dnf -y copr enable naccyde/criterion 
+          sudo dnf -y install cmake libbpf-devel \
+                              criterion-devel clang-tools-extra lcov libasan libubsan \
+                              doxygen python3-sphinx python3-breathe python3-furo
+      - name: Check out repository code
+        uses: actions/checkout@v3
+      - name: Create build directory
+        run: |
+          mkdir build
+          cd build
+      - name: Configure build
+        run: cmake -DCMAKE_BUILD_TYPE=${{ matrix.buildtype }} $GITHUB_WORKSPACE
```

Comment:
`cmake -B build -S $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=${{ matrix.buildtype }}` will create a build directory (`build`), and run CMake with sources from `$GITHUB_WORKSPACE`.